Projects/Claims/Sources/Views/SubmitClaim/LoadingViewWithState.swift

```diff
@@ -41,3 +43,81 @@ public struct LoadingViewWithState<Content: View, LoadingView: View, ErrorView:
         .presentableStoreLensAnimation(.easeInOut)
     }
 }
+
+public struct LoadingViewWithContent<Content: View>: View {
+    var content: () -> Content
+    @PresentableStore var store: ClaimsStore
+    private let action: ClaimsLoadingType
+
+    @State var presentError = false
+    @State var error = ""
+    @State var isLoading = false
+    var disposeBag = DisposeBag()
+
+    public init(
+        _ action: ClaimsLoadingType,
+        @ViewBuilder content: @escaping () -> Content
+    ) {
+        self.action = action
+        self.content = content
+    }
+    public var body: some View {
+        PresentableStoreLens(
+            ClaimsStore.self,
+            getter: { state in
+                state.loadingStates
+            }
+        ) { loadingStates in
+            ZStack {
+                content()
+                    .alert(isPresented: $presentError) {
+                        Alert(
+                            title: Text(L10n.somethingWentWrong),
+                            message: Text(error),
+                            dismissButton: .default(Text(L10n.alertOk))
+                        )
+                    }
+                if isLoading {
+                    HStack {
+                        WordmarkActivityIndicator(.standard)
+                    }
+                    .frame(maxWidth: .infinity, maxHeight: .infinity)
+                    .background(hBackgroundColor.primary.opacity(0.7))
+                    .cornerRadius(.defaultCornerRadius)
+                    .edgesIgnoringSafeArea(.top)
+                }
+            }
+        }
+        .presentableStoreLensAnimation(.easeInOut)
+        .onAppear {
```

Comment:
Here you could actually do `onReceive(store.stateSignal.publisher)` I think, then you don't have to worry about the DisposeBag and stuff 😄

Reply:
Unfortunately we can not do that since stateSignal is read only :(